src/components/NostoProduct/store.ts

```diff
@@ -40,41 +49,50 @@ export function createStore(element: NostoProduct) {
   }
 
   function selectSkuId(skuId: string) {
-    state.selectedSkuId = skuId
-    notify("selectedSkuId", skuId)
+    notify("selectedSkuId", (state.selectedSkuId = skuId))
+    const sku = state.skuData?.find(sku => sku.id === skuId)
+    if (sku) {
+      setSkuData(sku)
+    }
   }
 
   function selectSkuOption(optionId: string, skuIds: string[]) {
     state.skuOptions[optionId] = skuIds
     notify("skuOptions", state.skuOptions)
     const totalSelection = Object.keys(state.skuOptions).length
+    const selectedSkuIds = intersectionOf(...Object.values(state.skuOptions))
 
-    if (totalSelection === state.optionGroupCount) {
-      const selectedSkuIds = intersectionOf(...Object.values(state.skuOptions))
-      if (selectedSkuIds.length === 1) {
+    if (selectedSkuIds.length === 1) {
+      if (totalSelection === state.optionGroupCount) {
         selectSkuId(selectedSkuIds[0])
+      } else if (state.skuData) {
```

Comment:
Is it possible for sku selection to happen without `skuData` already set?

Reply:
yes, and that's supported